staging/src/k8s.io/apiserver/pkg/server/egressselector/egress_selector.go

```diff
@@ -36,13 +36,18 @@ import (
 	utilnet "k8s.io/apimachinery/pkg/util/net"
 	"k8s.io/apiserver/pkg/apis/apiserver"
 	egressmetrics "k8s.io/apiserver/pkg/server/egressselector/metrics"
+	compbasemetrics "k8s.io/component-base/metrics"
 	"k8s.io/component-base/tracing"
 	"k8s.io/klog/v2"
 	client "sigs.k8s.io/apiserver-network-proxy/konnectivity-client/pkg/client"
 )
 
 var directDialer utilnet.DialFunc = http.DefaultTransport.(*http.Transport).DialContext
 
```

Comment:
/assign @logicalhan 

to make sure this matches expected use and does sensible things when we are/aren't using network-proxy and uds config

Reply:
When I send a PR to pick up the new metrics, I'll make sure to have @logicalhan review. It can be unconditional, or conditional on use of [ProtocolGRPC](https://pkg.go.dev/k8s.io/apiserver@v0.25.4/pkg/apis/apiserver#ProtocolType) - I'm open to best practice here.